### DISCUSSION TITLE
In micronaut core 4.7.5 application,  annotation processor micronaut-inject-java has ClassNotFoundException

### DISCUSSION THREAD

**Discussion by ethanhere** [micronaut-projects/micronaut-core#11362] [Q&A]
With micronaut platform 4.7.0 (micronaut core 4.7.5) application (use maven and jdk21), I found that annotation processor micronaut-inject-java has ClassNotFoundException of io.micronaut.core.annotation.BuildTimeInit, or io.micronaut.context.conditions, or io.micronaut.core.annotation.AnnotationClassValue$UnresolvedClass

The build section of pom.xml
    <build>
        <plugins>

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>${version.maven.plugin.compiler}</version>

                <configuration>
                    <encoding>UTF-8</encoding>
                    <compilerArgs>
                        <arg>-parameters</arg>
                    </compilerArgs>
                    <annotationProcessorPaths>
                        <path>
                            <groupId>org.projectlombok</groupId>
                            <artifactId>lombok</artifactId>
                            <version>${1.18.32}</version>
                        </path>
                        <path>
                            <groupId>io.micronaut</groupId>
                            <artifactId>micronaut-inject-java</artifactId>
                            <version>${4.7.5}</version>
                        </path>
>                        <path>
>                            <groupId>io.micronaut</groupId>
>                            <artifactId>micronaut-context</artifactId>
>                            <version>${4.7.5}</version>
>                        </path>
                        <path>
                            <groupId>io.micronaut.micrometer</groupId>
                            <artifactId>micronaut-micrometer-annotation</artifactId>
                            <version>${5.7.1}</version>
                        </path>
                    </annotationProcessorPaths>
                </configuration>

                <executions>
                    <execution>
                        <id>test-compile</id>
                        <goals>
                            <goal>testCompile</goal>
                        </goals>
                        <configuration>
                            <compilerArgs>
                                <arg>-parameters</arg>
                            </compilerArgs>
                            <annotationProcessorPaths>
                                <path>
                                    <groupId>org.projectlombok</groupId>
                                    <artifactId>lombok</artifactId>
                                    <version>${1.18.32}</version>
                                </path>
                                <path>
                                    <groupId>io.micronaut</groupId>
                                    <artifactId>micronaut-inject-java</artifactId>
                                    <version>${4.7.5}</version>
                                </path>
>                                <path>
>                                    <groupId>io.micronaut</groupId>
>                                    <artifactId>micronaut-core</artifactId>
>                                    <version>${4.7.5}</version>
>                                </path>
                                <path>
                                    <groupId>io.micronaut.validation</groupId>
                                    <artifactId>micronaut-validation-processor</artifactId>
                                    <version>4.7.0</version>
                                </path>
                                <path>
                                    <groupId>io.micronaut.micrometer</groupId>
                                    <artifactId>micronaut-micrometer-annotation</artifactId>
                                    <version>${5.7.1}</version>
                                </path>
                            </annotationProcessorPaths>
                        </configuration>
                    </execution>
                </executions>
            </plugin>

When I use micronaut platform 4.5.1, I did not have micronaut-core and micronaut-context in annotationProcessorPaths. When I update to micronaut platform 4.7.0, I start to get compilation error of ClassNotFoundException (exception stack trace provided in PS section), and I add micronaut-core and micronaut-context in annotationProcessorPaths, the build works.

I like to know if this is the right way to fix the build when updating to 4.7.0? Or there is a "cleaner" way to handle this.

PS stack trace

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.1:compile (default-compile) on project microservice-framework-runtime-aws-generic: Compilation failure
[ERROR] java.lang.NoClassDefFoundError: io/micronaut/core/annotation/BuildTimeInit
[ERROR] 	at io.micronaut.inject.writer.EvaluatedExpressionProcessor.<clinit>(EvaluatedExpressionProcessor.java:49)
[ERROR] 	at io.micronaut.inject.writer.BeanDefinitionWriter.<init>(BeanDefinitionWriter.java:772)
[ERROR] 	at io.micronaut.inject.writer.BeanDefinitionWriter.<init>(BeanDefinitionWriter.java:653)
[ERROR] 	at io.micronaut.inject.processing.DeclaredBeanElementCreator.createBeanDefinitionVisitor(DeclaredBeanElementCreator.java:97)
[ERROR] 	at io.micronaut.inject.processing.DeclaredBeanElementCreator.buildInternal(DeclaredBeanElementCreator.java:82)
[ERROR] 	at io.micronaut.inject.processing.AbstractBeanElementCreator.build(AbstractBeanElementCreator.java:74)
[ERROR] 	at io.micronaut.annotation.processing.BeanDefinitionInjectProcessor.process(BeanDefinitionInjectProcessor.java:188)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1021)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:937)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1265)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1380)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1272)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:946)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:319)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:178)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:64)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:50)
[ERROR] Caused by: java.lang.ClassNotFoundException: io.micronaut.core.annotation.BuildTimeInit
[ERROR] 	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
[ERROR] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:593)
[ERROR] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
[ERROR] 	... 17 more

[ERROR] java.lang.NoClassDefFoundError: io/micronaut/core/annotation/AnnotationClassValue$UnresolvedClass
[ERROR] 	at io.micronaut.annotation.processing.JavaAnnotationMetadataBuilder.readAnnotationRawValues(JavaAnnotationMetadataBuilder.java:378)
[ERROR] 	at io.micronaut.annotation.processing.JavaAnnotationMetadataBuilder.readAnnotationRawValues(JavaAnnotationMetadataBuilder.java:358)
[ERROR] 	at io.micronaut.annotation.processing.JavaAnnotationMetadataBuilder.readAnnotationRawValues(JavaAnnotationMetadataBuilder.java:68)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.createAnnotationValue(AbstractAnnotationMetadataBuilder.java:899)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.lambda$annotationMirrorToAnnotationValue$5(AbstractAnnotationMetadataBuilder.java:849)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
[ERROR] 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.addAnnotations(AbstractAnnotationMetadataBuilder.java:987)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.addAnnotations(AbstractAnnotationMetadataBuilder.java:830)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.buildInternalMulti(AbstractAnnotationMetadataBuilder.java:800)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.buildInternal(AbstractAnnotationMetadataBuilder.java:246)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.lookupOrBuild(AbstractAnnotationMetadataBuilder.java:235)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.lookupOrBuildForType(AbstractAnnotationMetadataBuilder.java:199)
[ERROR] 	at io.micronaut.inject.ast.annotation.AbstractElementAnnotationMetadataFactory.lookupForClass(AbstractElementAnnotationMetadataFactory.java:202)
[ERROR] 	at io.micronaut.inject.ast.annotation.AbstractElementAnnotationMetadataFactory$11.lookup(AbstractElementAnnotationMetadataFactory.java:385)
[ERROR] 	at io.micronaut.inject.ast.annotation.AbstractElementAnnotationMetadataFactory$AbstractElementAnnotationMetadata.getCacheEntry(AbstractElementAnnotationMetadataFactory.java:451)
[ERROR] 	at io.micronaut.inject.ast.annotation.AbstractElementAnnotationMetadataFactory$AbstractElementAnnotationMetadata.getAnnotationMetadata(AbstractElementAnnotationMetadataFactory.java:461)
[ERROR] 	at io.micronaut.core.annotation.AnnotationMetadataDelegate.findRepeatableAnnotation(AnnotationMetadataDelegate.java:661)
[ERROR] 	at io.micronaut.core.annotation.AnnotationMetadataDelegate.findRepeatableAnnotation(AnnotationMetadataDelegate.java:661)
[ERROR] 	at io.micronaut.inject.annotation.AnnotationMetadataHierarchy.findRepeatableAnnotation(AnnotationMetadataHierarchy.java:1035)
[ERROR] 	at io.micronaut.core.annotation.AnnotationMetadata.hasStereotype(AnnotationMetadata.java:1497)
[ERROR] 	at io.micronaut.core.annotation.AnnotationMetadataDelegate.hasStereotype(AnnotationMetadataDelegate.java:519)
[ERROR] 	at io.micronaut.core.annotation.AnnotationMetadataDelegate.hasStereotype(AnnotationMetadataDelegate.java:519)
[ERROR] 	at io.micronaut.validation.visitor.ValidationVisitor.visitMethod(ValidationVisitor.java:102)
[ERROR] 	at io.micronaut.validation.visitor.ValidationVisitor.lambda$visitClass$0(ValidationVisitor.java:81)
[ERROR] 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
[ERROR] 	at io.micronaut.validation.visitor.ValidationVisitor.visitClass(ValidationVisitor.java:81)
[ERROR] 	at io.micronaut.annotation.processing.TypeElementVisitorProcessor.visitClass(TypeElementVisitorProcessor.java:316)
[ERROR] 	at io.micronaut.annotation.processing.TypeElementVisitorProcessor.process(TypeElementVisitorProcessor.java:269)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1021)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:937)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1265)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1380)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1272)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:946)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:319)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:178)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:64)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:50)
[ERROR] Caused by: java.lang.ClassNotFoundException: io.micronaut.core.annotation.AnnotationClassValue$UnresolvedClass
[ERROR] 	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
[ERROR] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:593)
[ERROR] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
[ERROR] 	... 47 more

**Comment by dstepanov** [comment]
Your build is misconfigured, it's recommended to use the version provided in BOM

**Reply by ethanhere** [reply]
@dstepanov could you please elaborate which version above was wrong?

**Reply by dstepanov** [reply]
It looks like you are using jars of Micronaut modules of incompatible versions

**Reply by ethanhere** [reply]
I double checked the versions and aligned to https://github.com/micronaut-projects/micronaut-platform/blob/v4.7.0/gradle/libs.versions.toml :
        <version.lombok>1.18.34</version.lombok>
  
        <version.micronaut.platform-bom>4.7.0</version.micronaut.platform-bom>
        <version.micronaut.core>4.7.5</version.micronaut.core>
        <version.micronaut.security>4.11.1</version.micronaut.security>
        <version.micronaut.aws>4.8.0</version.micronaut.aws>
        <version.micronaut.cache>5.1.0</version.micronaut.cache>
        <version.micronaut.data>4.10.3</version.micronaut.data>
        <version.micronaut.micrometer>5.9.2</version.micronaut.micrometer>
        <version.micronaut.validation>4.8.0</version.micronaut.validation>

I still have class not found exception if not add micronaut-core and micronaut-context to annotationProcessorPaths:
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.1:testCompile (test-compile) on project microservice-framework-runtime-aws-generic: Compilation failure: Compilation failure: 
[ERROR] ../MyObject.java:[13,42] error: cannot find symbol
[ERROR]   symbol:   class Builder
[ERROR]   location: class MyObject
[ERROR] java.lang.NoClassDefFoundError: io/micronaut/core/annotation/AnnotationClassValue$UnresolvedClass
[ERROR] 	at io.micronaut.annotation.processing.JavaAnnotationMetadataBuilder.readAnnotationRawValues(JavaAnnotationMetadataBuilder.java:378)
[ERROR] 	at io.micronaut.annotation.processing.JavaAnnotationMetadataBuilder.readAnnotationRawValues(JavaAnnotationMetadataBuilder.java:358)
[ERROR] 	at io.micronaut.annotation.processing.JavaAnnotationMetadataBuilder.readAnnotationRawValues(JavaAnnotationMetadataBuilder.java:68)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.createAnnotationValue(AbstractAnnotationMetadataBuilder.java:899)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.lambda$annotationMirrorToAnnotationValue$5(AbstractAnnotationMetadataBuilder.java:849)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
[ERROR] 	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
[ERROR] 	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
[ERROR] 	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.addAnnotations(AbstractAnnotationMetadataBuilder.java:987)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.addAnnotations(AbstractAnnotationMetadataBuilder.java:830)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.buildInternalMulti(AbstractAnnotationMetadataBuilder.java:800)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.buildInternal(AbstractAnnotationMetadataBuilder.java:246)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.lookupOrBuild(AbstractAnnotationMetadataBuilder.java:235)
[ERROR] 	at io.micronaut.inject.annotation.AbstractAnnotationMetadataBuilder.lookupOrBuildForType(AbstractAnnotationMetadataBuilder.java:199)
[ERROR] 	at io.micronaut.inject.ast.annotation.AbstractElementAnnotationMetadataFactory.lookupForClass(AbstractElementAnnotationMetadataFactory.java:202)
[ERROR] 	at io.micronaut.inject.ast.annotation.AbstractElementAnnotationMetadataFactory$11.lookup(AbstractElementAnnotationMetadataFactory.java:385)
[ERROR] 	at io.micronaut.inject.ast.annotation.AbstractElementAnnotationMetadataFactory$AbstractElementAnnotationMetadata.getCacheEntry(AbstractElementAnnotationMetadataFactory.java:451)
[ERROR] 	at io.micronaut.inject.ast.annotation.AbstractElementAnnotationMetadataFactory$AbstractElementAnnotationMetadata.getAnnotationMetadata(AbstractElementAnnotationMetadataFactory.java:461)
[ERROR] 	at io.micronaut.core.annotation.AnnotationMetadataDelegate.findRepeatableAnnotation(AnnotationMetadataDelegate.java:661)
[ERROR] 	at io.micronaut.core.annotation.AnnotationMetadataDelegate.findRepeatableAnnotation(AnnotationMetadataDelegate.java:661)
[ERROR] 	at io.micronaut.inject.annotation.AnnotationMetadataHierarchy.findRepeatableAnnotation(AnnotationMetadataHierarchy.java:1035)
[ERROR] 	at io.micronaut.core.annotation.AnnotationMetadata.hasStereotype(AnnotationMetadata.java:1497)
[ERROR] 	at io.micronaut.core.annotation.AnnotationMetadataDelegate.hasStereotype(AnnotationMetadataDelegate.java:519)
[ERROR] 	at io.micronaut.core.annotation.AnnotationMetadataDelegate.hasStereotype(AnnotationMetadataDelegate.java:519)
[ERROR] 	at io.micronaut.validation.visitor.ValidationVisitor.visitMethod(ValidationVisitor.java:102)
[ERROR] 	at io.micronaut.validation.visitor.ValidationVisitor.lambda$visitClass$0(ValidationVisitor.java:81)
[ERROR] 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
[ERROR] 	at io.micronaut.validation.visitor.ValidationVisitor.visitClass(ValidationVisitor.java:81)
[ERROR] 	at io.micronaut.annotation.processing.TypeElementVisitorProcessor.visitClass(TypeElementVisitorProcessor.java:316)
[ERROR] 	at io.micronaut.annotation.processing.TypeElementVisitorProcessor.process(TypeElementVisitorProcessor.java:269)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1021)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:937)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1265)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1380)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1272)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:946)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:319)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:178)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:64)
[ERROR] 	at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:50)
[ERROR] Caused by: java.lang.ClassNotFoundException: io.micronaut.core.annotation.AnnotationClassValue$UnresolvedClass
[ERROR] 	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
[ERROR] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:593)
[ERROR] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
[ERROR] 	... 47 more

**Reply by dstepanov** [reply]
Could you check the annotation processor dependencies? Also, why is the platform different from the core? Honestly, you should AVOID specifying the hardcoded versions and use the platform BOM!

**Reply by ethanhere** [reply]
according to  https://github.com/micronaut-projects/micronaut-platform/blob/v4.7.0/gradle/libs.versions.toml , micronaut-platform v4.7.0 uses micronaut-core 4.7.5, is that right? Are you suggesting to remove the version line from
 ```
                           <path>
                                <groupId>io.micronaut</groupId>
                                <artifactId>micronaut-inject-java</artifactId>
                                <version>${4.7.5}</version>
                            </path>

```
?

to
```
                            <path>
                                <groupId>io.micronaut</groupId>
                                <artifactId>micronaut-inject-java</artifactId>
                            </path>

```

**Reply by ethanhere** [reply]
I removed the version in all io.micronaut annotation process path items.
If still keep micronaut-context and micronaut-core, I got compilation error:
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.12.1:compile (default-compile) on project : Resolution of annotationProcessorPath dependencies failed: The following artifacts could not be resolved: io.micronaut:micronaut-context:jar:4.8.0 (absent): io.micronaut:micronaut-context:jar:4.8.0 was not found in https://maven

I don't know why micronaut-platform 4.7.0 bom tells maven to load micronaut-context 4.8.0. The greatest version of it is 4.7.5.

Then I removed micronaut-context and micronaut-core from annotation process path (also no version for io.micronaut:micronaut-inject-java etc, I got error
[ERROR] error: Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: Provider io.micronaut.annotation.processing.TypeElementVisitorProcessor could not be instantiated

**Reply by sdelamo** [reply]
as @dstepanov suggested we strongly recommend you use a single version number, [the platform-bom version](https://github.com/micronaut-projects/micronaut-platform/releases). The rest of the compatible versions are provided for you automatically. do you have a sample app which reproduces your issue?

**Reply by ethanhere** [reply]
I used micronaut platform 4.7.0, micronaut core 4.7.5. as I checked, all versions aligns with micronaut platform 4.7.0's version file.
I don't have a sample app to reproduce this issue because it may take us some time to work on it.
